tests/cupyx_tests/scipy_tests/signal_tests/test_spectral.py

```diff
@@ -1229,3 +1230,193 @@ def test_roundtrip_scaling(self, xp, scp):
             Zp0, input_onesided=True, boundary=True, scaling='psd')[1]
         results.append(x1)
         return results
+
+
+@testing.with_requires('scipy')
+class TestVectorstrength:
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_single_1dperiod(self, xp, scp):
+        events = xp.array([.5])
+        period = 5.
+
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_single_2dperiod(self, xp, scp):
+        events = xp.array([.5])
+        period = [1, 2, 5.]
+
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_equal_1dperiod(self, xp, scp):
+        events = xp.array([.25, .25, .25, .25, .25, .25])
+        period = 2
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_equal_2dperiod(self, xp, scp):
+        events = xp.array([.25, .25, .25, .25, .25, .25])
+        period = [1, 2, ]
+
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_spaced_1dperiod(self, xp, scp):
+        events = xp.array([.1, 1.1, 2.1, 4.1, 10.1])
+        period = 1
+
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_spaced_2dperiod(self, xp, scp):
+        events = xp.array([.1, 1.1, 2.1, 4.1, 10.1])
+        period = [1, .5]
+
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_partial_1dperiod(self, xp, scp):
+        events = xp.array([.25, .5, .75])
+        period = 1
+
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_partial_2dperiod(self, xp, scp):
+        events = xp.array([.25, .5, .75])
+        period = [1., 1., 1., 1.]
+
+        strength, phase = scp.signal.vectorstrength(events, period)
+        return strength, phase
+
+    @pytest.mark.xfail(
+        condition=os.name == 'linux' and testing.installed('scipy>=1.9.0'),
```

Comment:
Seems that `os.name` returns `posix` on linux (or at least Ubuntu on my local). Would you use `sys.platform` instead?